socorro/external/postgresql/setupdb_app.py

```diff
@@ -410,6 +346,85 @@ def create_table(element, compiler, **kw):
             db.set_table_owner(self.config.database_username)
             db.set_default_owner(database_name, self.config.database_username)
             db.set_grants(self.config)  # config has user lists
+        return True
+
+    def main(self):
+        database_name = self.config.database_name
+
+        if not database_name:
+            self.config.logger.error('"database_name" cannot be an empty string')
+            return 1
+
+        # superuser credentials for overall database
+        superuser_pg_url = self.create_connection_url(
+            'postgres',
+            self.config.database_superusername,
+            self.config.database_superuserpassword
+        )
+
+        # superuser credentials for working database
+        superuser_normaldb_pg_url = self.create_connection_url(
+            database_name,
+            self.config.database_superusername,
+            self.config.database_superuserpassword
+        )
+
+        # normal user credentials
+        normal_user_pg_url = self.create_connection_url(
+            database_name,
+            self.config.database_username,
+            self.config.database_password
+        )
+
+        # ensure that if on Heroku the the normal_user_pg_url and the
+        # superuser_pg_url are the same
+        if self.config.on_heroku and (normal_user_pg_url != superuser_pg_url):
+            self.config.logger.error(
+                'there is no superuser (%s) when using Heroku',
+                self.config.database_superusername
+            )
+            return 1
+
+        # table logging section
+        if self.config.unlogged:
+            @compiles(CreateTable)
+            def create_table(element, compiler, **kw):
+                text = compiler.visit_create_table(element, **kw)
+                text = re.sub("^\sCREATE(.*TABLE)",
+                              lambda m: "CREATE UNLOGGED %s" %
+                              m.group(1), text)
+                return text
+
+        # Postgres version check section
+        self.config.logger.info('Postgres version check section with %s', superuser_pg_url)
+        with PostgreSQLAlchemyManager(
+            superuser_pg_url,
+            self.config.logger,
+            autocommit=False
+        ) as db:
+            if not db.min_ver_check(90200):
+                self.config.logger.error(
+                    'unrecognized PostgreSQL version: %s',
+                    db.version_string()
+                )
+                self.config.logger.error('Only 9.2+ is supported at this time')
+                return 1
+
+        # At this point, we're done with setup and we can perform actions requested by the user
```

Comment:
Prior to this line, everything is identical to what was in `main`--nothing changed.